mqemitter-mongodb.js

```diff
@@ -59,66 +107,41 @@ function MQEmitterMongoDB (opts) {
   this._hasStream = false
   this._started = false
 
-  function waitStartup() {
-    that._collection = that._db.collection(opts.collection)
-    
-      that._collection.isCapped(function (err, capped) {
-        if (that.closed) { return }
-
-        if (err) {
-          // if it errs here, the collection might not exist
-          that._db.createCollection(opts.collection, {
-            capped: true,
-            size: opts.size,
-            max: opts.max
-          }, setLast)
-        } else if (!capped) {
-          // the collection is not capped, make it so
-          that._db.command({
-            convertToCapped: opts.collection,
-            size: opts.size,
-            max: opts.max
-          }, setLast)
-        } else {
-          setLast()
-        }
-      })
+  function waitStartup () {
+    checkCollection(that, setLast)
   }
 
   const oldEmit = MQEmitter.prototype.emit
 
   this._waiting = new Map()
   this._queue = []
   this._executingBulk = false
-  var failures = 0
+  let failures = 0
 
-  function setLast() {    
+  async function setLast () {
     try {
-    that._collection
-      .find({}, { timeout: false })
-      .sort({ $natural: -1 })
-      .limit(1)
-      .next(function (err, doc) {
-        if (err) {
-          that.status.emit('error', err)
-        }
-
-        that._lastObj = doc ? doc : { _id: new mongodb.ObjectID() };
-
-        if (!that._lastObj._stringId) {
-          that._lastObj._stringId = that._lastObj._id.toString()
-        }
-
-        start()
-      });
+      const results = await that._collection
+        .find({}, { timeout: false })
+        .sort({ $natural: -1 })
+        .limit(1)
+        .toArray()
+      const doc = results[0]
+      that._lastObj = doc || { _id: newId() }
+
+      if (!that._lastObj._stringId) {
+        that._lastObj._stringId = that._lastObj._id.toString()
+      }
+
+      await start()
     } catch (error) {
       that.status.emit('error', error)
     }
   }
 
-  function start() {
+  async function start () {
```

Comment:
Note that there is a setTimeout(start) underneath that would require some changes in the error handling.